examples/sprite_sheets/combine_images_into_sprite_sheet/combine_images_into_sprite_sheet.lpr

```diff
@@ -1,52 +1,118 @@
+  Type 0 if your spritesheet contains only one animation
+  Type a string as below if spritesheet contains multiple animations:
+  walk,0,9-run,10,19-idle,20,29
+  or
+  walk,1,10-run,11,20-idle,21,30
+  where the first parameter will be the name of the animation, the second and third parameter
+  the index of the first and last image of the animation
+  Each animation must be separated by the "-"
+  In the first case, only one animation, it will take the same name as the image (without "_x")
+  If you want to give the animation a different name instead of using the parameter "0" 
+  use only one set of parameters as in the case of multiple animations
+  e.g. with "walking_1.png" as the first image and 30 images in total: NewAnimName,1,30
+  
+
+  - 3rd parameter is the is the padding you used
+  For example:
+  Type 1 if the first image uses the form _0 or _1
+  Type 3 if the first image uses the form _000 or _001
+
+  - 4nd parameter is output image name
+
+  - 5rd parameter is the number of columns (how many frames in a row)
+
+  - 6rd parameter is 1 or 0 (True or False).
+    Use 1 to create the xml file that sets the (0, 0) coordinate of the spritesheet at the top left
+    (required by sprite-sheet-to-x3d), use 0 to sets the (0,0) coordinate at the bottom left.
+
+  Some examples:
+  (under Windows use .\)
+  One animation, "walking_1.png" as the first image:
+  ./combine_images_into_sprite_sheet walking_1.png 0 1 sprite_sheet_24fps.png 6 1
+  Three animation, "left_00.png" as first image
+  ./combine_images_into_sprite_sheet ../sprite/left_00.png walk,0,9-run,10,19-idle,20,29 2 ../sprite/sprite_sheet_10fps.png 3 1
 
   See compile_and_run.sh script to real-life example.
 }
 
 program combine_images_into_sprite_sheet;
 
-uses SysUtils, Math,
+uses SysUtils, Math, Classes,
   CastleParameters, CastleImages, CastleVideos, CastleUtils, CastleVectors;
 
 var
-  InputUrl, OutputUrl: String;
+  InputUrl, AnimOpt, Ext, OutputUrl, AnimName, AnimNameIndex: String;
+  FilePath, FileName, FileExt, FileNameOutExt: string;
   InputVideo: TVideo;
   OutputImage: TRGBAlphaImage;
-  Columns, FinalColumns, Rows, I, X, Y: Cardinal;
+  Columns, Padding, FinalColumns, Rows, I, X, Y, K, A, H, from, unto: Cardinal;
+  ReverseY, SingleElement, StartFromZero: boolean;
+  XMLFileList: TStringList;
+  XMLFileName: String;
+  ElementList, AnimList, CheckPadding: TStringArray;
+
 begin
-  Parameters.CheckHigh(3);
+  Parameters.CheckHigh(6);
   InputUrl := Parameters[1];
-  OutputUrl := Parameters[2];
-  Columns := StrToInt(Parameters[3]);
+  AnimOpt := Parameters[2];
+  Padding := StrToInt(Parameters[3]);
+  OutputUrl := Parameters[4];
+  Columns := StrToInt(Parameters[5]);
+  ReverseY := StrToBool(Parameters[6]);
 
   try
+    FilePath := ExtractFilePath(InputUrl);
+    FileName := ExtractFileName(InputUrl);
+    FileExt := ExtractFileExt(InputUrl);
+    FileNameOutExt := ChangeFileExt(FileName, '');
```

Comment:
Operate on URLs using routines from `CastleURIUtils`, do not assume they are filenames.

So ExtractFileName should be ExtractURIName (it actually does the same *for now*). Same for other file-name operations.